### ISSUE TITLE
Warning and Hint Pragmas do not print to console when declared from a std lib module

### ISSUE THREAD

**Issue by rayman22201** [nim-lang/Nim#12424]
`{.Warning.}` and `{.Hint.}` pragmas that are declared inside std lib modules are ignored and not displayed to the user when compiling.

The same pragmas in user code display correctly during compilation.

Note:
`{.error.}` and `{.deprecated.}` work correctly in std lib modules.

According to @krux02:
> it probably has to do with the fact that many messages from the standard library are prohibited. Pretty annoying if you ask me. It should be to not bother user with internal warning of the standard library, but it also prevents me from actually seeing them in the first place so they can be fixed.

### Example
Here is an example of why this is not a good design:

This PR had to be reverted because it broke behavior for some users:
https://github.com/nim-lang/Nim/pull/12231

TLDR; the PR added a `{.error.}` pragma to the `locks.nim` module if `threads` are not enabled.

I attempted to re-implement the PR using a `{.Warning.}` pragma as a compromise.
Unfortunately, converting the message to a `{.Warning.}` caused Nim to completely ignore the message.

This is a completely valid case for a std lib module providing a useful warning to the user, and not just noise from the std lib internals.

**Comment by krux02** [comment]
I would also like to mention that it became annoying for me when working for the std lib module because this behavior actively hid warnings problems from me that I was trying to find and fix.

**Comment by rayman22201** [comment]
I would like to note that I am game for doing the work to change this behavior, but I need help finding where to start. I just did a quick search, but I was not able to find the code where this is happening.

**Comment by Araq** [comment]
In ``compiler/modules.nim`` there is this piece of code:

```nim

  graph.config.notes =
    if s.owner.id == graph.config.mainPackageId or isDefined(graph.config, "booting"): graph.config.mainPackageNotes
    else: graph.config.foreignPackageNotes
```

So you can enable these already via ``-d:booting`` but of course a better solution would be nice.

**Comment by rayman22201** [comment]
Two ideas:

1.) Get rid of `-d:booting` and just always return `graph.config.mainPackageNotes`
Seems like what @krux02 would like. But idk the consequences

2.) Create a new pragma like `{.externalWarning: "string"}` that is enabled in `graph.config.foreignPackageNotes`

I don't love the idea of adding a new pragma, but maybe that's the safest thing to do?

Thoughts?

**Comment by disruptek** [comment]
//cc @narimiran

**Comment by ghost** [comment]
Fixed by https://github.com/nim-lang/Nim/pull/17024, needs a test case I guess.
